providers/flagd/pkg/provider.go

```diff
@@ -158,6 +161,12 @@ func (p *Provider) ObjectEvaluation(
 	return p.service.ResolveObject(ctx, flagKey, defaultValue, evalCtx)
 }
 
+func (p *Provider) setStatus(status of.State) {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+	p.status = status
```

Comment:
I would assume this is atomic and therefore locking doesn't help? :thinking:

Reply:
Though I think you want to make sure this can run DURING init.

I think I thread isn't blocked by a lock it already owns, right? So I guess you might want to lock in init() and well as here.

Reply:
I think we do need it, yes - if you execute the tests with the `-race` flag enabled, a data race error occurs, due to `p.status` being set in the goroutine created within the `Init()` function, while being accessed via the `Status()` function in the test